lib/diagnostic.rb

```diff
@@ -6,35 +6,42 @@ class Diagnostic
   # a Fixnum.
   def longest(list)
     # Your code goes here
+    list.max
   end
 
   # Question 2
   # Write ruby code to return the sum of entries in the
   # list supplied (i.e. apply the `+` operator).
-  # Assume each entry implements a `+` method.
+  # Assume each entry implements  a `+` method.
   def sum(list)
     # Your code goes here
+    sum = 0
+    list.each.to_i { |i| sum += i }
   end
 
   # Question 3
   # Write ruby code that returns true if some of the entries
   # in the list of numbers are odd
   def some_odd(numbers)
     # Your code goes here
+   numbers.any? { |x| x.to_i % 2 != 0}
   end
 
-  # Question 4
+
+  # Question 4d
   # Write ruby code that returns true if every entry in the
   # list of numbers are even
   def every_even(numbers)
     # Your code goes here
+     numbers.all? { |x| x.to_i % 2 == 0 }
   end
 
   # Question 5
   # Write ruby code that returns an array with all of the
   # entries from the list of words capitalized (i.e. 'foo' becomes 'Foo').
   def transform(words)
     # Your code goes here
+    words.first.upcase
```

Comment:
Close `words.map(&:capitalize)`. Remember that `.map` re-writes each element of the array to what every you're doing to the element.